src/metabase/models/setting.clj

```diff
@@ -98,6 +104,18 @@
                   (tru "Setting {0} does not exist.\nFound: {1}" k (sort (keys @registered-settings)))))))))
 
 
+(defn- call-on-change
+  "Cache watcher that applies `:on-change` callback for all settings that have changed."
+  [_key _ref old new]
+  (let [rs @registered-settings]
+    (doseq [changed-setting (-> (data/diff old new)
+                                (second)
+                                (keys))]
+      (when-let [on-change (get-in rs [(keyword changed-setting) :on-change])]
+        (on-change (clojure.core/get old changed-setting) (clojure.core/get new changed-setting))))))
```

Comment:
does it make sense to call `on-change` if the `old` and `new` values are the same?

Reply:
`on-change` will only get called if the values are different